libr/debug/p/native/linux/linux_debug.c

```diff
@@ -494,7 +533,7 @@ static void linux_add_and_attach_new_thread(RDebug *dbg, int tid) {
 	} else {
 		tid_info = r_debug_pid_new ("new_path", tid, uid, 's', 0);
 	}
-	(void) linux_attach_single_pid (dbg, tid);
```

Comment:
The new child is attached at this point but it has a pending SIGSTOP and we can't affect it until it arrives. We should waitpid here, you might have added more sigstop handling elsewhere because this wasn't handled properly.

Reply:
This is handled in the `linux_dbg_wait` which will look for `SIGTRAP` in other threads when it hits a `SIGSTOP` caused by newly created thread/process. I could probably move that handling to here because iirc there is a weird bug forces me to do it in `linux_dbg_wait`. I will check it again.

Reply:
This is actually checked in r_debug_native_continue for SIGTRAP sent with `si_pid == 0` when attaching.

Reply:
`linux_add_and_attach_new_thread` is only used in `linux_ptrace_event` when `PTRACE_EVENT_CLONE` happens to add the new thread into the thread list. The initial `SIGSTOP` is handled in `linux_ptrace_event` or by `waitpid` depending on whether `waitpid` got the `SIGTRAP` from the parent or the initial `SIGSTOP` from the child first.